src/classes/RD2_ScheduleService.cls

```diff
@@ -81,6 +90,174 @@ public without sharing class RD2_ScheduleService {
         return newSchedules;
     }
 
+    /***
+    * @description Synchronizes the schedule records to match an updated recurring donation
+    * @param rd Recurring Donation record
+    * @param schedules Recurring Donation Schedule record
+    * @return TDTM_Runnable.DMLWrapper
+    */
+    public TDTM_Runnable.DMLWrapper getUpdatedSchedules(npe03__Recurring_Donation__c rd, List<RecurringDonationSchedule__c> schedules) {
```

Comment:
I like that this single method returns back both the new and updated schedules, though I haven't really figured out if DMLWrapper is the right construct. It clearly works and it is the only data type we have in the package to meet this objective. And, it's certainly fine for now. Let's talk about it later this week or next though to see if there's an alternative approach that doesn't feel so much like it's tied to TDTM. There may not be, but it's probably worth the hypothetical discussion.